signac/core/validators.py

```diff
@@ -0,0 +1,99 @@
+# Copyright (c) 2017 The Regents of the University of Michigan
+# All rights reserved.
+# This software is licensed under the BSD 3-Clause License.
+"""Validators for SyncedCollection API."""
+import warnings
+from collections.abc import Mapping
+from collections.abc import Sequence
+
+from ..errors import InvalidKeyError
+from ..errors import KeyTypeError
+from .synced_collection import NUMPY
+
+if NUMPY:
+    import numpy
+
+
+def NoDotInKey(data):
+    """Raise an exception if there is dot in key.
+
+    Parameters
+    ----------
+    data:
```

Comment:
See #362 - I think the colon is unnecessary unless a data type is specified.
```suggestion
    data
```